api/client/build.go

```diff
@@ -324,6 +312,24 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	return nil
 }
 
+// validateTag checks if the given image name can be resolved.
+func validateTag(rawRepo string) (string, error) {
```

Comment:
And I think the func signature is imposed by `opts.NewListOpts` (the `(string, error)` returns).